tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -917,122 +1041,133 @@ TEST_CASE("Test Tap position optimizer") {
                 SUBCASE("no line drop compensation") { check_b = test::check_exact_per_strategy(3, 4, 2); }
                 SUBCASE("resistance") {
                     update_data.line_drop_compensation_r = 0.5 / base_power_3p;
-                    check_b = test::check_exact_per_strategy(3, 5, 3);
+                    check_b = test::check_compensated_exact_per_strategy(3, 3, 3, 3, 4, 1);
                 }
                 SUBCASE("positive reactance") {
                     update_data.line_drop_compensation_x = 0.125 / base_power_3p;
-                    check_b = test::check_exact_per_strategy(3, 5, 2);
+                    check_b = test::check_compensated_exact_per_strategy(3, 3, 4, 2, 4, 1);
                 }
                 SUBCASE("negative reactance") {
                     update_data.line_drop_compensation_x = -0.5 / base_power_3p;
-                    check_b = test::check_exact_per_strategy(3, 5, 3);
+                    check_b = test::check_compensated_exact_per_strategy(3, 3, 3, 3, 4, 1);
                 }
 
                 regulator_b.update(update_data);
-            }
 
-            SUBCASE("multiple transformers with control function based on ranking") {
-                state_a.rank = 0;
-                state_b.rank = 1;
-                state_a.tap_min = -5;
-                state_a.tap_max = 5;
-                state_b.tap_min = -5;
-                state_b.tap_max = 5;
-
-                state_a.u_pu = [&state_a, &regulator_a](ControlSide side) {
-                    CHECK(side == regulator_a.control_side());
-
-                    // u_2a = f(tap_pos_a) when rank is 0
-                    // u_2a = (u_1a * n_1) / (1.0 + relative_tap_pos_a)
-                    // consider u_1a = n_1 = 1.0
-                    // For a tap_size of 0.1 and tap_nom of 0, tap_pos_relative_a = 0.1 * (tap_pos_a - 0)
-                    auto const relative_tap_a = static_cast<double>(state_a.tap_pos) * 0.1;
-                    return static_cast<DoubleComplex>(1.0 / (1.0 + relative_tap_a));
-                };
-
-                state_b.u_pu = [&state_a, &regulator_a, &state_b, &regulator_b](ControlSide side) {
-                    CHECK(side == regulator_b.control_side());
+                SUBCASE("multiple transformers with control function based on ranking") {
```

Comment:
did you deliberately move this subcase into the negative reactance subcase`?
```suggestion
            }
            SUBCASE("multiple transformers with control function based on ranking") {
```

Reply:
When debugging, some brackets have to be relocated ;)

Reply:
please fix this before merging